ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.tsx

```diff
@@ -59,48 +67,54 @@ const ConfirmAddCustodianToken = () => {
   }, [connectRequest, history, mostRecentOverviewPage]);
 
   const handleButtonClick = useCallback(
-    async ({ isConfirm }) => {
+    async (isConfirm: boolean) => {
       try {
         if (isConfirm) {
           setConnectError('');
           setIsLoading(true);
 
-          if (connectRequest.chainId) {
+          if (connectRequest?.chainId) {
             const networkType = Object.keys(BUILT_IN_NETWORKS).find(
               (key) =>
-                Number(BUILT_IN_NETWORKS[key].chainId).toString(10) ===
-                connectRequest.chainId.toString(),
-            );
-
-            await dispatch(setProviderType(networkType));
+                Number(
+                  BUILT_IN_NETWORKS[key as keyof typeof BUILT_IN_NETWORKS]
+                    .chainId,
+                ).toString(10) === connectRequest.chainId.toString(),
+            ) as NetworkType | undefined;
+
+            if (networkType) {
+              await dispatch(setProviderType(networkType));
+            }
           }
         }
 
-        await dispatch(
-          mmiActions.removeAddTokenConnectRequest({
-            origin: connectRequest.origin,
-            environment: connectRequest.environment,
-            token: connectRequest.token,
-          }),
-        );
-
-        trackEvent({
-          category: MetaMetricsEventCategory.MMI,
-          event: MetaMetricsEventName.TokenAdded,
-          properties: {
-            actions: isConfirm
-              ? 'Custodian RPC confirm'
-              : 'Custodian RPC cancel',
-            custodian: connectRequest.custodian,
-            envName: connectRequest.environment,
-          },
-        });
-
-        if (isConfirm) {
-          history.push(CUSTODY_ACCOUNT_ROUTE);
+        if (connectRequest) {
```

Comment:
This seems like a new logic branch. Ideally, these typescript migration PRs shouldn't include logic changes.

Reply:
Not really. Because of the typescript migration, I added this safety check, but If you think it doesn't make sense to be in this PR, I can remove it and add it in another PR. Let me now, please :)